var/spack/repos/builtin/packages/bison/package.py

```diff
@@ -51,7 +51,8 @@ class Bison(AutotoolsPackage, GNUMirrorPackage):
     depends_on('help2man', type='build')
 
     patch('pgi.patch', when='@3.0.4')
-    patch('nvhpc.patch', when='%nvhpc')
+    patch('nvhpc-3.6.patch', when='@3.6.0:3.6.99 %nvhpc')
+    patch('nvhpc-3.7.patch', when='@3.7.0:3.7.99 %nvhpc')
```

Comment:
Can you briefly comment above what the patches do and why this is necessary? Also, bonus if you can delete the patches from the repository in favor of something retrieved remotely.

Reply:
The Nvidia compilers currently do not support some of these builtins.  Since fallback paths are available in the code, these changes just add the condition to detect that the Nvidia compiler is being used.  

The goal is to implement these builtins in a future compiler release and make these patches or any upstream changes unnecessary.

Reply:
@samcmill Can you add these explanations in form of a comment above the respective patches?

Reply:
Added